src/multi_dict.jl

```diff
@@ -66,6 +67,46 @@ function in(pr::(Tuple{Any,Any}), d::MultiDict{K,V}) where {K,V}
     (v !== Base.secret_table_token) && (pr[2] in v)
 end
 
+# TODO: For efficiency, we probably want a MultiKeySet to correspond to Dict's KeySet
+
+# TODO: I'm not actually sure if we need this eachindex?
+function Base.eachindex(md::MultiDict)
+    flatten((repeated(k, length(vs)) for (k,vs) in md.d))
+end
+Base.length(md::MultiDict) = sum(length(vs) for (_,vs) in md.d)
+# TODO: what should values() do?
+Base.values(md::MultiDict) = values(md.d)
+#function Base.values(md::MultiDict)
+#    flatten(vs for (_,vs) in md.d)
+#end
+
```

Comment:
One remaining question is what should `values(md::MultiDict)` return? Should it iterate all of the (flattened) values? Or should it iterate over the vectors?

That is, which of these options is most consistent and useful:
```julia
julia> values(MultiDict(3=>3, 1=>2, 1=>4))
Base.ValueIterator for a Dict{Int64,Array{Int64,1}} with 2 entries. Values:
  [3]
  [2, 4]
```
or
```julia
julia> values(MultiDict(3=>3, 1=>2, 1=>4))
Base.ValueIterator for a MultiDict{Int64,Int64} with 3 entries. Values:
  3
  2
  4
```

For consistency with `iterate()` I would think it should be the latter. But for consistency with `getindex`, I could see an argument for the former.

In fact, for consistency with `getindex`, I can understand why iterate might return what it did before this PR!
I'm starting to wonder if maybe we actually want to provide like two different types of interfaces over this dict? One for the aggregated view and one for the pairs view?

But, I guess, if you just want the aggregated view, you could just use a `DefaultDict{K, Vector{V}}`... So i think continuing to pursue the pairs-based interface makes sense.

Reply:
FWIW, the `golang` multimap does indeed flatten the values (the second approach), despite its `Get(k)` function returning a vector. So I think we should do this same. this is what felt most intuitive to me.

https://github.com/jwangsadinata/go-multimap/blob/c29f3d7f33b68d8bcd82880a558e9f8eed58e2b3/setmultimap/setmultimap.go#L144-L154